File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Trackers/DoneDone/DoneDoneDetail.cs
namespace bugTrapKit.DoneDone
{
	public enum DoneDoneDetail
	{
		Images,
		Project,
		Title,
		Description,
		Priority,
		Fixer,
		Tester,
		DueDate,
		Tags,
		Notify
	}

	public static class DoneDoneDetailExtensions
	{
		public static string SettingsKey (this DoneDoneDetail detail)
		{
			switch (detail) {
			case DoneDoneDetail.Images:
				return "DoneDoneDetailImages";
			case DoneDoneDetail.Project:
				return "DoneDoneDetailProject";
			case DoneDoneDetail.Title:
				return "DoneDoneDetailTitle";
			case DoneDoneDetail.Description:
				return "DoneDoneDetailDescription";
			case DoneDoneDetail.Priority:
				return "DoneDoneDetailPriority";
			case DoneDoneDetail.Fixer:
				return "DoneDoneDetailFixer";
			case DoneDoneDetail.Tester:
				return "DoneDoneDetailTester";
			case DoneDoneDetail.DueDate:
				return "DoneDoneDetailDueDate";
			case DoneDoneDetail.Tags:
				return "DoneDoneDetailTags";
			case DoneDoneDetail.Notify:
				return "DoneDoneDetailNotify";
			default:
				return null;
			}
		}

		public static bool IsPersonDetail (this DoneDoneDetail detail)
		{
			switch (detail) {
			case DoneDoneDetail.Fixer:
			case DoneDoneDetail.Tester:
			case DoneDoneDetail.Notify:
				return true;
			default:
				return false;
			}
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Trackers/DoneDone/DoneDoneDetail.swift
//
//  DoneDoneDetail.swift
//  bugTrap
//
//  Created by Colby L Williams on 11/14/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import Foundation

enum DoneDoneDetail: Int, TrackerDetailType {
    case Images = 0, Project, Title, Description, Priority, Fixer, Tester, DueDate, Tags, Notify

    func isPersonDetail () -> Bool {
        switch self {
        case .Images, .Project, .Title, .Description, .Priority, .Tags, .DueDate: return false
        case .Fixer, .Tester, Notify: return true
        }
    }

	var settingsKey: String {
		switch self {
            case .Images: return "DoneDoneDetailImages"
            case .Project: return "DoneDoneDetailProject"
            case .Title: return "DoneDoneDetailTitle"
            case .Description: return "DoneDoneDetailDescription"
            case .Priority: return "DoneDoneDetailPriority"
            case .Fixer: return "DoneDoneDetailFixer"
            case .Tester: return "DoneDoneDetailTester"
            case .DueDate: return "DoneDoneDetailDueDate"
            case .Tags: return "DoneDoneDetailTags"
            case .Notify: return "DoneDoneDetailNotify"
		}
	}
}